src/app/modules/myinfo/myinfo.util.ts

```diff
@@ -285,16 +286,23 @@ export const createRelayState = (formId: string): string =>
  * Validates that a form is a MyInfo form with an e-service ID
  * @param form Form to validate
  */
-export const validateMyInfoForm = (
-  form: IFormSchema | IPopulatedForm,
-): Result<IMyInfoForm, MyInfoNoESrvcIdError | MyInfoAuthTypeError> => {
+export const validateMyInfoForm = <T extends IFormSchema>(
+  form: T,
+): Result<MyInfoForm<T>, FormAuthNoEsrvcIdError | AuthTypeMismatchError> => {
   if (!form.esrvcId) {
-    return err(new MyInfoNoESrvcIdError())
+    return err(new FormAuthNoEsrvcIdError(form._id))
   }
-  if (form.authType !== AuthType.MyInfo) {
-    return err(new MyInfoAuthTypeError())
+  if (isMyInfoForm(form)) {
+    return ok(form)
   }
-  return ok(form as IMyInfoForm)
+  return err(new AuthTypeMismatchError(AuthType.MyInfo, form.authType))
+}
+
+// Typeguard to ensure that form has eserviceId and MyInfo authType
+const isMyInfoForm = <F extends IFormSchema>(
+  form: F,
+): form is MyInfoForm<F> => {
+  return form.authType === AuthType.MyInfo && !!form.esrvcId
```

Comment:
Doesn't seem like it, but ensure that the additional `!!form.esrvcId` check is not a regression!

Reply:
good catch! will do so (also for the below one)

Reply:
on prod, we don't allow the eserviceId field to be empty. this would get caught by the first layer of checks (and the correct error returned). hence, this isn't a regression!